File: app/src/main/java/com/xattacker/android/bingo/logic/BingoGrid.java
package com.xattacker.android.bingo.logic;

import com.xattacker.android.bingo.logic.BingoLogic.PlayerType;

public interface BingoGrid
{
	void initial();
	
	PlayerType getType();
	void setType(PlayerType aType);
	
	int getValue();
	void setValue(int aValue);
	
	boolean isSelected();
	void setSelected(boolean aSelected);
	
	boolean isConnected();
	void setConnected(boolean aConnected);
	
	boolean isLineConnected(ConnectedDirection aDirection);
	void setConnectedLine(ConnectedDirection aDirection, boolean aConnected);
}

File: app/src/main/java/com/xattacker/android/bingo/logic/BingoGrid.kt
package com.xattacker.android.bingo.logic

interface BingoGrid
{
    var type: PlayerType
    var value: Int
    var isSelectedOn: Boolean // could not use name isSelected: it will cause accidental override with class View
    var isConnected: Boolean

    operator fun get(direction: ConnectedDirection): Boolean
    operator fun set(direction: ConnectedDirection, connected: Boolean)

    fun initial()
}
